src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -188,12 +206,12 @@ CHIP_ERROR OnMessageToSend(TransferSession::OutputEvent & event)
         if (err != CHIP_NO_ERROR) {
             mExchangeCtx->Close();
             mExchangeCtx = nullptr;
-            ResetState();
-        } else if (event.msgTypeData.HasMessageType(Protocols::SecureChannel::MsgType::StatusReport)) {
+            ResetState(err);
+        } else if (msgTypeData.HasMessageType(Protocols::SecureChannel::MsgType::StatusReport)) {
             // If the send was successful for a status report, since we are not expecting a response the exchange context is
             // already closed. We need to null out the reference to avoid having a dangling pointer.
             mExchangeCtx = nullptr;
-            ResetState();
+            ResetState(CHIP_ERROR_INTERNAL); // TODO: is this an error scenario?
```

Comment:
This is a "we decided to send an error status for some reason" scenario.  In theory, we could have extracted the exact status we sent from the event.MsgData (before we moved it into SendMessage), but not sure it's worth it.